front/lib/models.ts

```diff
@@ -832,6 +832,64 @@ ChatRetrievedDocument.init(
 
 ChatMessage.hasMany(ChatRetrievedDocument);
 
+export class TrackedDocument extends Model<
+  InferAttributes<TrackedDocument>,
+  InferCreationAttributes<TrackedDocument>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+
+  declare documentId: string;
+  declare trackingEnabledAt: Date | null;
+  declare email: string;
+
+  declare dataSourceId: ForeignKey<DataSource["id"]>;
+}
+
+TrackedDocument.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    documentId: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    trackingEnabledAt: {
+      type: DataTypes.DATE,
+      allowNull: true,
+      defaultValue: DataTypes.NOW,
+    },
+    email: {
```

Comment:
Then if we do a strong email matching, I thiiiink we can have a userId here instead of email?

Reply:
Yes def